### DISCUSSION TITLE
Multi Module Support

### DISCUSSION THREAD

**Discussion by rasfarrf5** [FilledStacks/stacked#957] [General]
Currently stacked architecture supports a single module. But flutter already enabled multiple modules.

The current problem is, we can't move the Router (generated file) outside of other packages. If possible to extract then we can move all views to its dedicated module and use the router to go from one view to another.

**Comment by FilledStacks** [comment]
@rasfarrf5 this is something that we have brought up before. There is a workaround that we've used before but it's not as clean as we'd like it to be. 

How are you currently handling your multi module requirements?

**Reply by rasfarrf5** [reply]
Currently I extracted only locator to different module. But the Router is still keeping under main module. 

Can you help to share the approach with the workaround. It might be helpful for my project.
Thanks in advance. 👍🏼

**Reply by FilledStacks** [reply]
It's not really easy to share the approach because we might not have the same setup. The reason we had to do multi module split is because we had 2 applications that shared a lot of the same code, but had to be different apps. 

So I created a core package that has all the shared views and UI in it. That core package also exposes the coreLoctor + coreRoutes. 
When the app starts We add the routes from the app entry point and the routes from the coreRouter together and then run the app. This means we can use the same navigation service through out the app to navigate to a view in core or in the main contiainer entry point app. 

It just means we had an app.dart in our core package and in our main package (only with UI specific to the main package).

**Reply by rasfarrf5** [reply]
Got it, I found an approach like core locator and core router. Now I'm able to extract all views to dedicated modules. 
Thanks for the suggestion.
